### ISSUE TITLE
CronWorkflow with multiple steps

### ISSUE THREAD

**Issue by matheusjericoo** [argoproj/argo#2524]
# Summary

How I can do cronworkflow with two steps (DAG)?

# Motivation

I need create a workflow to machine learning pipeline (preprocess and training) in two different container.
I did workflow work in "Kind: Workflow" (code bellow), but i need schedule every day this workflow. Than, i think that CronJobs is solution for my problem.

How I tranform Workflow to CronWorkflow?

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: ml-workflow-argo
spec:
  entrypoint: ml-workflow-dag
  templates:
  - container:
      image: "gcr.io/{project_id}/{name_image}:{tag}"
      command: ['python', '/scripts/data.py']
    name: preprocess

  - container:
      image: "gcr.io/{project_id}/{name_image):{tag}"
      command: ['python', '/scripts/train.py']
    name: training

  - dag:
      tasks:
      - name: preprocess-step
        template: preprocess
      - name: training-step
        template: training
        dependencies:
        - preprocess-step
    name: ml-workflow-dag
```

I did, but didn't work:
(template-argo.yaml)
```yaml
apiVersion: argoproj.io/v1alpha1
kind: CronWorkflow
metadata:
  name: ml-workflow-argo
spec:
  schedule: "* * * * *"
  concurrencyPolicy: "Allow"
  successfulJobsHistoryLimit: 4
  failedJobsHistoryLimit: 4     
  suspend: false
  workflowSpec:
    entrypoint: ml-workflow-dag
    templates:
      - name: preprocess
        container:
          image: "gcr.io/{project_id}/{name_image}:{tag}"
          command: ['python', '/scripts/data.py']

      - name: training
        container:
          image: "gcr.io/{project_id}/{name_image}:{tag}"
          command: ['python', '/scripts/train.py']

      - dag:
          tasks:
          - name: preprocess-step
            template: preprocess
          - name: training-step
            template: training
            dependencies:
            - preprocess-step
        name: ml-workflow-dag
```

```bash
argo submit --watch template-argo.yaml
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/argoproj/argo/cmd/argo/commands.waitOrWatch(0x0, 0x0, 0x0, 0x0, 0x0, 0x10100)
	/root/go/src/github.com/argoproj/argo/cmd/argo/commands/submit.go:142 +0xb1
github.com/argoproj/argo/cmd/argo/commands.SubmitWorkflows(0xc4203bb080, 0x1, 0x2, 0xc420469a80, 0xc4203baf20)
	/root/go/src/github.com/argoproj/argo/cmd/argo/commands/submit.go:116 +0x810
github.com/argoproj/argo/cmd/argo/commands.NewSubmitCommand.func1(0xc420345680, 0xc4203bb080, 0x1, 0x2)
	/root/go/src/github.com/argoproj/argo/cmd/argo/commands/submit.go:40 +0x5a
github.com/argoproj/argo/vendor/github.com/spf13/cobra.(*Command).execute(0xc420345680, 0xc4203bb040, 0x2, 0x2, 0xc420345680, 0xc4203bb040)
	/root/go/src/github.com/argoproj/argo/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/argoproj/argo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420257b80, 0xc420344f00, 0xc420344c80, 0xc420344a00)
	/root/go/src/github.com/argoproj/argo/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/argoproj/argo/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420257b80, 0xc4203badc0, 0xc4204a5bf0)
	/root/go/src/github.com/argoproj/argo/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/root/go/src/github.com/argoproj/argo/cmd/argo/main.go:15 +0x27

```

**Comment by sarabala1979** [comment]
you need to create cronworkflow  using `argo cron create <filename>`
```
manage cron workflows

Usage:
  argo cron [flags]
  argo cron [command]

Available Commands:
  create      create a cron workflow
  delete      delete a cron workflow
  get         display details about a cron workflow
  lint        validate files or directories of cron workflow manifests
  list        list cron workflows
  resume      resume zero or more cron workflows
  suspend     suspend zero or more cron workflows
```

**Comment by simster7** [comment]
@matheusjericoo The answer above is correct. Could you let us know if it works? Also, what is your `argo version`? I want to fix the panic error

**Comment by matheusjericoo** [comment]
Argo version:
```sh 
$ argo version

argo: v2.2.1
  BuildDate: 2018-10-11T16:26:28Z
  GitCommit: 3b52b26190163d1f72f3aef1a39f9f291378dafb
  GitTreeState: clean
  GitTag: v2.2.1
  GoVersion: go1.10.3
  Compiler: gc
  Platform: linux/amd64
```

I runned the command, but don't worked:
```sh
$ argo cron create template-argo.yaml

Error: unknown command "cron" for "argo"
Run 'argo --help' for usage.
unknown command "cron" for "argo"

```

**Comment by simster7** [comment]
@matheusjericoo `CronWorkflows` are only available on `v2.5+`. You are running `v2.2.1`, so you'll need to upgrade before you can use them.

**Comment by matheusjericoo** [comment]
It's worked!! Thanks
